File: content/05/code/ruby/10-changed-other-scenario-and-deleted-unused-stepdef/features/step_definitions/steps.rb
require 'shouty'

DEFAULT_RANGE = 100

Before do
  @people = {}
  @network = Shouty::Network.new(DEFAULT_RANGE)
end

Given "the range is {int}" do |range|
  @network = Shouty::Network.new(range)
end

Given "a person named {word}" do |name|
  @people[name] = Shouty::Person.new(@network, 0)
end

Given "people are located at" do |table|
  table.symbolic_hashes.each do |name: , location: |
    @people[name] = Shouty::Person.new(@network, location.to_i)
  end
end

When "Sean shouts" do
  @people["Sean"].shout("Hello, world")
end

When "Sean shouts {string}" do |message|
  @people["Sean"].shout(message)
  @message_from_sean = message
end

Then "Lucy should hear Sean's message" do
  expect(@people['Lucy'].messages_heard).to eq [@message_from_sean]
end

Then "Lucy should hear a shout" do
  expect(@people['Lucy'].messages_heard.count).to eq 1
end

Then "Larry should not hear Sean's message" do
  expect(@people['Larry'].messages_heard).not_to include(@message_from_sean)
end

Then "Larry should not hear a shout" do
  expect(@people['Larry'].messages_heard.count).to eq 0
end

File: content/05/code/js/10-changed-other-scenario-and-deleted-unused-stepdef/features/step_definitions/steps.js
const { Given, When, Then, Before } = require('cucumber')
const { assertThat, contains, is, not } = require('hamjest')

const { Person, Network } = require('../../src/shouty')

const default_range = 100

Before(function () {
  this.people = {}
  this.network = new Network(default_range)
})

Given('the range is {int}', function (range) {
  this.network = new Network(range)
})

Given('a person named {word}', function (name) {
  this.people[name] = new Person(this.network, 0)
})

Given('people are located at', function (dataTable) {
  dataTable.hashes().map((person) => {
    this.people[person.name] = new Person(this.network, person.location)
  })
})

When('Sean shouts', function () {
  this.people['Sean'].shout('Hello, world')
})

When('Sean shouts {string}', function (message) {
  this.people['Sean'].shout(message)
  this.messageFromSean = message
})

Then('Lucy should hear Sean\'s message', function () {
  assertThat(this.people['Lucy'].messagesHeard(), contains(this.messageFromSean))
})

Then('Lucy should hear a shout', function () {
  assertThat(this.people['Lucy'].messagesHeard().length, is(1))
})

Then('Larry should not hear Sean\'s message', function () {
  assertThat(this.people['Larry'].messagesHeard(), not(contains(this.messageFromSean)))
})

Then('Larry should not hear a shout', function () {
  assertThat(this.people['Larry'].messagesHeard().length, is(0))
})
